modules/login/src/main/resources/META-INF/resources/login/js/registration.js

```diff
@@ -0,0 +1,401 @@
+        termsUrl = url;
+    }
+
+    function init() {
+
+        if (!bInitialized) {
+            //Selector cache
+            $formStep1 = $('#formStep1');
+            $formStep2 = $('#formStep2');
+            $formStep3 = $('#formStep3');
+            $formStep4 = $('#formStep4');
+
+            $dlgStep1 = $('#step1');
+            $dlgStep2 = $('#step2');
+            $dlgStep3 = $('#step3');
+            $dlgStep4 = $('#step4');
+            $dlgConfirmation = $('#regConfirmation');
+            $dlgError = $('#errordialog');
+
+            $selectOrgType = $('#selectOrgAffiliates');
+            $selectOrg = $('#selectOrganizations');
+
+            $selectImtCDF = $('#selectCDFIMT');
+            $selectImtFed = $('#selectFedIMT');
+            $selectImtLocal = $('#selectLocalIMT');
+            $selectImtUSAR = $('#selectUSAR');
+
+            $($formStep1).validator().on('submit', doStep2);
+
+            $($formStep2).validator().on('submit', doStep3);
+            //Listen for field validation events on Form2 in order to detect a valid email address
+            // was entered and send to the verify service.
+            //$($formStep2).validator().on('validated.bs.validator', form2OnValidateField);
+
+            $($formStep3).validator().on('submit', doStep4);
+            $($formStep4).validator().on('submit', doRegistration);
+            $($formStep1).on('reset', onResetForm1);
+            $($formStep2).on('reset', onResetForm2);
+            $($formStep3).on('reset', onResetForm3);
+            $($formStep4).on('reset', onResetForm4);
+
+
+            //Load the T&C content externally.
+            $('#tc_content').load(termsUrl + " " + termsElement);
+            $("#phone").mask("(999) 999-9999");
+
+            populateOrgData();
+            bInitialized = true;
+        }
+
+
+    }
+
+    function resetDialogSequence() {
+        $($formStep1)[0].reset();
+        $($formStep2)[0].reset();
+        $($formStep3)[0].reset();
+        $($formStep4)[0].reset();
+        $($dlgStep1).removeClass('hidden');
+        $($dlgStep2).addClass('hidden');
+        $($dlgStep3).addClass('hidden');
+        $($dlgStep4).addClass('hidden');
+        $($dlgConfirmation).addClass('hidden');
+        $($dlgError).addClass('hidden');
+
+    }
+    function onResetForm1() {
+        //Need to handle redraw of the BS Select dropdowns since only the bound <select> element.
+        // is reset.
+        setTimeout(function() {
+            $('.selectpicker', $formStep1).selectpicker('render');
+            $('#btnNext1').prop('disabled', 'disabled');
+        },0); //Set Timeout so that this code block doesn't get exec'd before the form reset happens.
+
+
+    }
+    function onResetForm2() {
+    }
+    function onResetForm3() {
+        setTimeout(function() {
+            $('.selectpicker', $formStep3).selectpicker('render');
+        },0); //Set Timeout so that this code block doesn't get exec'd before the form reset happens.
```

Comment:
This comment is repeated, indicating a pattern. Consider extracting a function that makes the pattern clear.